### PR TITLE
fix: make braintree process paypal payments as well

### LINKED ISSUE #6267: PayPal via Braintree is broken and dangerous!
### What I'm trying to achieve
Try to successfully checkout using the Braintree Gateway, Drop-In UI in the frontend and PayPal as payment method.

### Steps to reproduce the problem
1. Use Saleor 2.11, create a checkout with PayPal
2. Look at the Braintree dashboard - the amount gets captured successfully, but Saleor is failing critically! So it is impossible for the frontend to tell the user, that the checkout was actually successful.
3. The request is failing in Saleor, because `lower()` could not be run on type `none`:             `brand=credit_card.get("card_type", "").lower(),`

The `def authorize` function https://github.com/mirumee/saleor/blob/bb9ffe88f7275e2a52c801334f1e00bbfab2c1bf/saleor/payment/gateways/braintree/__init__.py#L124
is actually really! unsafe. When it fails out of any reason, there is no protection, like a refund of the already captured amount.
Furthermore, the backend is just prepared for credit cards. When using PayPal, there is no need to collect the brand of the credit card .. 

As we are using Braintree / Paypal heavily and are planing to integrate local payment providers like Klarna, we would like to discuss the implementation of braintree with you guys here and want to help out with merge requests. We created a temporary fix, that is at least a little more stable than the current approach, by using a feature from the braintree SDK: `braintree_sdk.PaymentInstrumentType.PayPalAccount` 

```python
   if(error):
        print('error occured')
        return GatewayResponse(
            is_success=result.is_success,
            action_required=False,
            kind=kind,
            amount=gateway_response.get("amount", payment_information.amount),
            currency=gateway_response.get("currency", payment_information.currency),
            transaction_id=gateway_response.get(
                "transaction_id", payment_information.token
            ),
            error=error,
            raw_response=gateway_response,
        )
    if(result.transaction.payment_instrument_type == braintree_sdk.PaymentInstrumentType.PayPalAccount):
        print('is paypal')
        return GatewayResponse(
            is_success=result.is_success,
            action_required=False,
            kind=kind,
            amount=gateway_response.get("amount", payment_information.amount),
            currency=gateway_response.get("currency", payment_information.currency),
            customer_id=gateway_response.get("customer_id"),
            transaction_id=gateway_response.get(
                "transaction_id", payment_information.token
            ),
            error=error,
            payment_method_info=PaymentMethodInfo(
                type="paypal",
            ),
            raw_response=gateway_response,
        )
    if(result.transaction.payment_instrument_type == braintree_sdk.PaymentInstrumentType.CreditCard):
        print('is creditcard')
        return GatewayResponse(
            is_success=result.is_success,
            action_required=False,
            kind=kind,
            amount=gateway_response.get("amount", payment_information.amount),
            currency=gateway_response.get("currency", payment_information.currency),
            customer_id=gateway_response.get("customer_id"),
            transaction_id=gateway_response.get(
                "transaction_id", payment_information.token
            ),
            error=error,
            payment_method_info=PaymentMethodInfo(
                last_4=credit_card.get("last_4"),
                exp_year=credit_card.get("expiration_year"),
                exp_month=credit_card.get("expiration_month"),
                brand=credit_card.get("card_type", "PayPal").lower(),
                name=credit_card.get("cardholder_name"),
                type="card",
            ),
            raw_response=gateway_response,
        )
    # unsafe TODO: uncapture payment
    return GatewayResponse(
        is_success=result.is_success,
        action_required=False,
        kind=kind,
        amount=gateway_response.get("amount", payment_information.amount),
        currency=gateway_response.get("currency", payment_information.currency),
        transaction_id=gateway_response.get(
            "transaction_id", payment_information.token
        ),
        error=error,
        raw_response=gateway_response,
    )
```

### PR DESCRIPTION
**Pull request by JannikZed**
I want to merge this change because...

this pull request enables the Braintree Gateway to successfully process PayPal payments as well. It closes #6267

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/payment/gateways/braintree/__init__.py [hunk 1]
```diff
@@ -141,28 +141,75 @@ def authorize(
         raise BraintreeException(DEFAULT_ERROR_MESSAGE)
 
     gateway_response = extract_gateway_response(result)
+
     error = get_error_for_client(gateway_response["errors"])
     kind = TransactionKind.CAPTURE if config.auto_capture else TransactionKind.AUTH
     credit_card = gateway_response.get("credit_card", {})
+
+    if(error):
```
Comment (maarcingebala) [inline saleor/payment/gateways/braintree/__init__.py:149]: The code should be formatted with Black. The recommended way is to use pre-commit which runs all linters automatically.

File: saleor/payment/gateways/braintree/__init__.py [hunk 2]
```diff
@@ -141,28 +141,75 @@ def authorize(
...
+        print('error occured')
```
Comment (maarcingebala) [inline saleor/payment/gateways/braintree/__init__.py:150]: Print should be removed before merging.

File: saleor/payment/gateways/braintree/__init__.py [hunk 3]
```diff
@@ -141,28 +141,75 @@ def authorize(
...
+        return GatewayResponse(
```
Comment (maarcingebala) [inline saleor/payment/gateways/braintree/__init__.py:151]: A lot of code seems to be duplicated when constructing the gateway response objects here. Maybe we could create first a gateway response with fields that are common to all cases, and then inside the `if` conditions only provide fields specific to these particular cases?

File: saleor/payment/gateways/braintree/__init__.py [hunk 4]
```diff
@@ -141,28 +141,75 @@ def authorize(
...
+                last_4=credit_card.get("last_4"),
+                exp_year=credit_card.get("expiration_year"),
+                exp_month=credit_card.get("expiration_month"),
+                brand=credit_card.get("card_type", "").lower(),
+                name=credit_card.get("cardholder_name"),
+                type="card",
+            ),
+            raw_response=gateway_response,
+        )
+    # unsafe TODO: uncapture payment
```
Comment (maarcingebala) [inline saleor/payment/gateways/braintree/__init__.py:202]: Can we add support for it now? If not, I would extend this TODO to explain what case exactly is unhandled / what needs to be done.

**Review by korycins — COMMENTED** [review]
@JannikZed  I believe that the issue is that we dummy assume that payment comes from card. Can't we do something like this?
```
def authorize(
    payment_information: PaymentData, config: GatewayConfig
) -> GatewayResponse:
    try:
        if not payment_information.customer_id:
            result = transaction_for_new_customer(payment_information, config)
        else:
            result = transaction_for_existing_customer(payment_information, config)
    except braintree_sdk.exceptions.NotFoundError:
        raise BraintreeException(DEFAULT_ERROR_MESSAGE)

    gateway_response = extract_gateway_response(result)
    error = get_error_for_client(gateway_response["errors"])
    kind = TransactionKind.CAPTURE if config.auto_capture else TransactionKind.AUTH

    credit_card = gateway_response.get("credit_card") or {}
    credit_card_type = credit_card.get("card_type") or "" 
    payment_method_type = str(result.transaction.payment_instrument_type)

    return GatewayResponse(
        is_success=result.is_success,
        action_required=False,
        kind=kind,
        amount=gateway_response.get("amount", payment_information.amount),
        currency=gateway_response.get("currency", payment_information.currency),
        customer_id=gateway_response.get("customer_id"),
        transaction_id=gateway_response.get(
            "transaction_id", payment_information.token
        ),
        error=error,
        payment_method_info=PaymentMethodInfo(
            last_4=credit_card.get("last_4"),
            exp_year=credit_card.get("expiration_year"),
            exp_month=credit_card.get("expiration_month"),
            brand=credit_card_type.lower(),
            name=credit_card.get("cardholder_name"),
            type=payment_method_type,
        ),
        raw_response=gateway_response,
    )
```
This should properly handle all types of payments.

**Comment by JannikZed** [comment]
already fixed by #7300

**Pull request closed without merging**